src/components/forms/EditGrid/EditGridItem.tsx

```diff
@@ -0,0 +1,191 @@
+  data?: never;
+  canEdit?: never;
+  saveLabel?: never;
+  onReplace?: never;
+}
+
+interface WithIsolation<T> {
+  /**
+   * In isolation mode, fields of an item can be edited without them affecting the
+   * entire form state. Only when the edits are saved/confirmed, is the parent form
+   * state updated.
+   */
+  enableIsolation: true;
+  /**
+   * Existing item data, passed to the underlying Formik component as initial state.
+   * Note that this is only accepted as *initial* data and changes in props do not change
+   * the form field values.
+   */
+  data: T;
+  /**
+   * Callback to render the main content of a single item. Gets passed options representing
+   * the current UI state.
+   */
+  getBody: (opts: {expanded: boolean}) => React.ReactNode;
+  /**
+   * If true, edit control button(s) are rendered.
+   */
+  canEdit?: boolean;
+  /**
+   * Custom label for the save/confirm button.
+   */
+  saveLabel?: string;
+  /**
+   * Callback invoked when confirming the item changes.
+   */
+  onReplace: (newValue: T) => void;
+}
+
+export type EditGridItemProps<T> = EditGridItemBaseProps & (WithoutIsolation | WithIsolation<T>);
+
+// TODO: track open/collapsed state and adapt the buttons accordingly
+
+function EditGridItem<T extends JSONObject = JSONObject>({
+  index,
+  heading,
+  canRemove,
+  removeLabel,
+  onRemove,
+  ...props
+}: EditGridItemProps<T>) {
+  const intl = useIntl();
+  const [expanded, setExpanded] = useState<boolean>(false);
+  const headingId = useId();
+
+  // TODO
+  const errors: any = {};
+  const zodSchema: any = null;
+  return (
+    <Fieldset className="openforms-editgrid__item">
+      {heading && (
+        <FieldsetLegend className="openforms-editgrid__item-heading" id={headingId}>
+          {heading}
+        </FieldsetLegend>
+      )}
+
+      {/*
+        In isolation mode, set up a separate Formik context that requires user interaction
+        before it's committed to the parent state. In non-isolation mode, the caller needs
+        to provide the form fields with their deeply nested field names so the state is
+        updated directly, so then we render as-is.
+      */}
+
+      {props.enableIsolation && props.canEdit ? (
+        // We apply the same Formik rendering philosophy as in FormioForm.tsx w/r to initial
+        // values, errors, touched state and the validation behaviour (validate individual fields, on blur)
+        <Formik<T>
+          initialValues={props.data}
+          initialErrors={{}}
+          initialTouched={false ? setNestedObjectValues(errors, true) : undefined}
+          // when removing items, the order changes, so we must re-render to display the
+          // correct data
+          enableReinitialize
+          validateOnChange={false}
+          validateOnBlur={false}
+          validationSchema={false ? toFormikValidationSchema(zodSchema) : undefined}
+          // TODO: trigger submit from primary button - perhaps Formik should wrap the entire thing?
```

Comment:
remove, it's handled

Reply:
indeed - didn't want to hit you with the "Refresh" link while reviewing :stuck_out_tongue: